backend/src/main/java/com/fake_orgasm/users_management/models/User.java

```diff
@@ -36,9 +43,56 @@ public class User implements Comparable<User> {
      * @throws NullPointerException if the specified object is null
      * @throws ClassCastException   if the specified object's type prevents it
      *                              from being compared to this object.
+     *
+     */
+    private int id;
+
+    private String name;
+    private String lastName;
+    private Date dateBirth;
+    private List<Integer> flights;
+    private Category category;
+    private String country;
+    /**
+     * This is the constructor of the class.
+     * <p>
+     * The constructor receives all the necessary parameters to correctly
+     * represent the User data structure within the program.
+     *
+     * @param id int receives the id of the user.
+     * @param name String receives the name of the user.
+     * @param lastNam String receives the last name of the user.
+     * @param dateBirth Date receives the date birth of the user.
+     * @param cat Category receives the category of the user.
+     * @param country String receives the country of the user.
+     */
+    public User(int id, String name, String lastNam, Date dateBirth, Category cat, String country) {
+        this.id = id;
+        this.name = name;
+        this.lastName = lastNam;
+        this.dateBirth = dateBirth;
+        this.category = cat;
+        this.country = country;
+    }
+
+    /**
+     * This method compare the actual user with other user.
+     * This method use the category as parameter of compare, return a positive
+     * value if the current user  has a higher category than the
+     * compared user, and return a negative value in the other case,
+     * return 0 when the user have the same category.
+     *
+     * @param o the object to be compared.
+     * @return int result of the comparative.
      */
     @Override
     public int compareTo(User o) {
```

Comment:
The comparison should be by the name and if  both have the same name, it should be by the id

Reply:
Solved